chore(docs): Algolia search guide from class to functional components

janosh opened the pull request:
## Description

This PR updates the Algolia search guide from class to functional components using React Hooks.

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which we can import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https: ...
 
-```shell
+```sh
````
Please keep `shell`

```suggestion
```shell
```
https://www.gatsbyjs.org/contributing/gatsby-style-guide/#format-code-blocks-inline-code-and-images

LekoArts commented on docs/docs/adding-search-with-algolia.md:
```diff
@@ -2,33 +2,39 @@
 ## Configuring the Algolia plugin
 
-First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia), [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) and [`algoliasearch`](https://github.com/algolia/algoliasearch-client-javascript) to your project. With `react-instantsearch` Algolia provides an extensive library of off-the-shelf React components that we can import to  ...
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which we can import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https: ...
```
```suggestion
First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which you can import to save ourselves a lot of work. You'll also be using `dotenv` which gets shipped with Gatsby by default. You're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
```

dotenv already comes with Gatsby. Also: https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun

DSchau replied:
@LekoArts it's a bit non-ideal to rely on downstream dependencies to exist (and be the expected version), so I'd keep the installation of dotenv.

LekoArts replied:
mh, interesting. We do the same (relying on that fact) everywhere else I think, e.g. in the env variables documentation.

janosh replied:
I'm happy to go either way but I agree with @DSchau that it's often good to keep dependencies explicit. Of course, consistency is nice as well.

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
 
-```shell
+```sh
 yarn add gatsby-plugin-algolia react-instantsearch-dom algoliasearch dotenv
````
```suggestion
npm install --save gatsby-plugin-algolia react-instantsearch-dom algoliasearch
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -40,32 +46,33 @@ module.exports = {
 Notice that we're loading `queries` from a file at `./src/utils/algolia.js` (you can of course put it wherever you like) and our Algolia ID and API key from `.env` so let's add those files.
 
-```env:title=.env
+```sh:title=.env
````
```suggestion
```text:title=.env
```

janosh replied:
I usually get better syntax highlighting (e.g. of comments) in `.env` files with `sh` but I assume `text` is convention?

LekoArts commented on docs/docs/adding-search-with-algolia.md:
```diff
@@ -98,7 +105,7 @@ const postQuery = `{
           title
           slug
-          date(formatString: "MMM DD, YYYY")
+          date(formatString: "MMM D, YYYY")
```
Why this change?

janosh replied:
Looks better in my opinion. Just a small oversight during the original writing of this guide but we can also keep it as is.

LekoArts replied:
Okay 👍

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
 ```
 
-Next, add `gatsby-plugin-algolia` to your `gatsby-config.js`.
+We will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
````
```suggestion
You will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
-Next, add `gatsby-plugin-algolia` to your `gatsby-config.js`.
+We will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
+
+```sh
````
```suggestion
```shell
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
+We will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
+
+```sh
+yarn add react styled-components gatsby-plugin-styled-components
````
```suggestion
npm install --save styled-components gatsby-plugin-styled-components
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -2,33 +2,39 @@
+yarn add react styled-components gatsby-plugin-styled-components
+```
+
+Next, add `gatsby-plugin-algolia` and `gatsby-plugin-styled-components` if you installed it to your `gatsby-config.js`.
````
```suggestion
Next, add `gatsby-plugin-algolia` and `gatsby-plugin-styled-components` to your `gatsby-config.js`.
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -132,152 +139,197 @@ const queries = [
+
+### `index.js`
+
+```jsx:title=src/components/search/index.js
````
```suggestion
```js:title=src/components/search/index.js
```

LekoArts commented on docs/docs/adding-search-with-algolia.md:
````diff
@@ -297,74 +349,17 @@ export default connectSearchBox(({ refine, ...rest }) => (
+## `styled.js`
 
-```js:title=src/components/Search/styles.js
+```jsx:title=src/components/search/styles.js
````
```suggestion
```js:title=src/components/search/styles.js
```

janosh commented:
@LekoArts Thanks a lot for making the change from "we" to "you". Sorry for the slow reaction. Was travelling with poor connection these past couple of days but I'm back now and just changed all `jsx` code blocks to `js` and added titles where they were missing.

LekoArts commented:
@janosh No need to apologize, thank you very much for your effort! :)
LGTM 👍
